core/src/main/resources/struts-default.xml

```diff
@@ -372,6 +373,10 @@
                  associated with requests that should not be validated.
                  -->
             <interceptor-stack name="defaultStack">
+                <interceptor-ref name="cspInterceptor">
+                    <param name="reporting">false</param>
```

Comment:
A few things to unpack here:

- Remember that you can have enable reporting for CSP on both enforce and report modes. It is important that we distinguish between 'you are setting your policy in report-mode' and 'you are setting an endpoint where reports will be sent'. 
- Having a key called `reporting` here is a little confusing, because it's not clear whether setting this to `false` makes the policy be in report mode or if it will add the `report-to` directive. 
- If I understand correctly, the current implementation **always** sets the `report-to` directive. This behaviour should be configurable, so we should only add `report-to` if a `reportUri` has been specified. 
- That being set, it might make more sense if we change this key's name to something along the lines of  `reporting-mode`.
- It would be nice to keep code and config consistent in terms of defaults. At the moment, `DefaultCspSettings` sets reporting (report mode) to `true`, but this config is setting it to `false`, which is strange. Let's align both of them by **enabling CSP in report-mode by default**, if we start enforcing from the get go, we won't give the dev a chance to find breakages and refactor, things will just break. 
- The previous also applies to `reportUri` (of which www.google.com is definitely not a value we want to set btw xD), let's align this `<param>` value to the default in the code. The value for `reportUri` should be a relative URL.
- Finally, I think we should probably move this interceptor slightly lower down in the default stack. Let's let the really core interceptors (like errors, servletConfig, etc.) run first. This will give us some guarantees that if a request hits our interceptor, it's already passed all sorts of basic checks (it's a well formed HTTP request, it maps to a servlet, it triggers no obvious exceptions, etc etc)

Reply:
Configured